src/react/RegisterDevicePage.js

```diff
@@ -336,6 +374,132 @@ export default class RegisterDevicePage extends React.Component {
       showNamespaceModal: false
     });
   }
+
+  toggleIntrospection(e) {
+    const sendIntrospection = e.target.checked;
+
+    this.setState({
+      sendIntrospection: sendIntrospection
+    });
+  }
+
+  addInterfaceToIntrospection(interfaceId) {
+    this.setState((state, props) => ({
+      introspectionInterfaces: state.introspectionInterfaces.set(interfaceId.name, interfaceId)
+    }));
+  }
+
+  removeIntrospectionInterface(interfaceId) {
+    this.setState((state, props) => {
+      let introspection = state.introspectionInterfaces;
+      introspection.delete(interfaceId.name);
+
+      return {
+        introspectionInterfaces: introspection
+      }
+    });
+  }
+}
+
+function InstrospectionTable({interfaces, onAddInterface, onRemoveInterface}) {
+  return (
+    <Table className="mb-4" responsive>
+      <thead>
+        <tr>
+          <th>Interface name</th>
+          <th>Major</th>
+          <th>Minor</th>
+          <th className="action-column"></th>
+        </tr>
+      </thead>
+      <tbody>
+        { Array.from(interfaces).map(([key, interfaceId]) =>
+          <InterfaceIntrospectionRow
+            key={key}
+            name={interfaceId.name}
+            major={interfaceId.major}
+            minor={interfaceId.minor}
+            onRemove={() => onRemoveInterface(interfaceId)}
+          />
+        )}
+        <IntrospectionControlRow
+          onAddInterface={onAddInterface}
+        />
+      </tbody>
+    </Table>
+  );
+}
+
+function InterfaceIntrospectionRow({name, major, minor, onRemove}) {
+  return (
+    <tr>
+      <td>{name}</td>
+      <td>{major}</td>
+      <td>{minor}</td>
+      <td>
+        <i className="fas fa-eraser color-red action-icon" onClick={onRemove} />
+      </td>
+    </tr>
+  );
+}
+
+function IntrospectionControlRow({onAddInterface}) {
+  const [name, setName] = useState("");
+  const [major, setMajor] = useState(0);
+  const [minor, setMinor] = useState(1);
+
+  return (
+    <tr>
+      <td>
+        <Form.Control
+          type="text"
+          placeholder="Interface name"
+          value={name}
+          onChange={(e) => {setName(e.target.value)}}
+        />
+      </td>
+      <td>
+        <Form.Control
+          type="number"
+          min="0"
+          value={major}
+          onChange={(e) => {setMajor(e.target.value)}}
```

Comment:
Be aware that e.target.value is a string, you should be saving a number instead. The same goes for the interface's minor